crates/noirc_evaluator/src/ssa/conditional.rs

```diff
@@ -400,8 +412,11 @@ impl DecisionTree {
         predicate: AssumptionId,
     ) {
         if predicate == AssumptionId::dummy() || self[predicate].value != Some(ctx.zero()) {
+            let mut short_circuit = false;
             for i in instructions {
-                self.conditionalise_into(ctx, result, *i, predicate);
+                if !self.conditionalise_into(ctx, result, *i, predicate, short_circuit) {
+                    short_circuit = true;
```

Comment:
Can add a `break` here

Reply:
No, it is important to continue to process the instructions in the short-circuit mode, which delete them.